src/cljs/main/quil_site/examples/fireworks.cljs

```diff
@@ -0,0 +1,133 @@
+      :velocity velocity
+      :location (add velocity location)
+      :lifespan (- lifespan 10)
+      :acceleration [0 0])))
+
+(defn update-bomb [{:keys [acceleration velocity location] :as bomb}]
+  (let [velocity (add velocity acceleration)]
+    (assoc bomb
+      :velocity velocity
+      :location (add velocity location)
+      :acceleration [0 0])))
+
+(defn apply-gravity-to-particles [fireworks gravity]
+  (update fireworks :particles #(map (fn [particle] (apply-force particle gravity)) %)))
+
+(defn update-particles [fireworks]
+  (update fireworks :particles #(map update-particle %)))
+
+(defn remove-dead-particles [fireworks]
+  (update fireworks :particles #(remove dead? %)))
+
+(defn apply-gravity-to-bombs [fireworks gravity]
+  (update fireworks :bombs #(map (fn [bomb] (apply-force bomb gravity)) %)))
+
+(defn update-bombs [fireworks]
+  (update fireworks :bombs #(map update-bomb %)))
+
+(defn add-bomb [fireworks]
+  (if (zero? (rem (q/frame-count) (* 5 (int (q/random 1 5)))))
+   (update fireworks :bombs #(conj % (create-bomb (:origin fireworks))))
+   fireworks))
+
+(defn remove-dead-bombs [fireworks]
+  (update fireworks :bombs (fn [bombs]
+                             (remove (fn [bomb]
+                                       (> (second (:velocity bomb)) 0)) bombs))))
+
+(defn explode-bomb [{:keys [lifespan location velocity] :as bomb}]
+  (when (> (second velocity) 0)
+    (map #(create-particle location velocity) (range 300))))
+
+(defn explode-bombs [fireworks]
+  (update fireworks :particles #(concat % (mapcat explode-bomb (:bombs fireworks)))))
+
+(defn draw-particle [{:keys [lifespan] [x y] :location :as particle}]
+  (let [s (q/map-range lifespan 0 255 0 (/ (q/width) 100))]
+    (q/fill (q/color 255 (- 255 lifespan) 0 lifespan))
+    (q/ellipse x y s s)))
+
+(defn draw-bomb [{:keys [location] :as bomb}]
+  (q/fill 35)
+  (let [s (/ (q/width) 100)
+        [x y] location]
+    (q/ellipse x y s s)))
+
+(defn setup []
+  (q/color-mode :rgb)
+  (q/blend-mode :add)
+  {:particles ()
+   :bombs ()
+   :origin [(/ (q/width) 2)
+            (q/height)]})
+
+(defn update-fireworks [fireworks]
+  (let [gravity [0 5]]
+      (-> fireworks
+          (apply-gravity-to-particles gravity)
+          (update-particles)
+          (remove-dead-particles)
+          (apply-gravity-to-bombs gravity)
+          (update-bombs)
+          (explode-bombs)
+          (remove-dead-bombs)
+          (add-bomb))))
+
+(defn draw-fireworks [{:keys [particles bombs] :as fireworks}]
+  (q/clear)
+  (q/background 0)
+  (doseq [bomb bombs]
+   (draw-bomb bomb))
+  (doseq [particle particles]
+    (draw-particle particle)))
+
+(defn run-sketch [host size] ;DELETE
+(q/sketch
+  :host host
+  :size [size size]
+  :setup setup
+  :update update-fireworks
+  :draw draw-fireworks
+  :middleware [m/fun-mode])
+) ;DELETE
+
+(quil-site.main/register-example! "fireworks" "Anthony Galea" run-sketch :interactive? true :white-play-button? true) ;DELETE
```

Comment:
I don't think it's interactive? Interactive means that mouse/keyboard is used to affect sketch and I don't see the sketch using any input events.

Reply:
Thanks, fixed.